src/js/rtp-stats.js

```diff
@@ -5,86 +5,123 @@
                         frameRateReceived:  when_defined(statsReport.googFrameRateReceived),
                         procMilliseconds:   is_defined(statsReport.googCurrentDelayMs),
                         jbMilliseconds:     when_defined(statsReport.googJitterBufferMs)
-                    };
+                    }, statsReport.type, streamType);
 
                 }
             } else if (statsReport.type === 'inboundrtp') {
                 // Firefox case. Firefox reports packetsLost parameter only in inboundrtp type, and doesn't report in outboundrtp type.
                 // So we only pull from inboundrtp. Firefox reports only stats for the stream passed in.
                 if (is_defined(statsReport.packetsLost) && is_defined(statsReport.packetsReceived)) {
-                    extractedStats = {
+                    extractedStats = new MediaRtpStats({
                         packetsLost:        statsReport.packetsLost,
                         packetsCount:       statsReport.packetsReceived,
                         audioLevel:         when_defined(statsReport.audioInputLevel),
-                        rttMilliseconds:    streamType === 'audio_ouptut' || streamType === 'video_output' ? when_defined(statsReport.roundTripTime) : null,
+                        rttMilliseconds:    streamType === 'audio_output' || streamType === 'video_output' ? when_defined(statsReport.roundTripTime) : null,
                         jbMilliseconds:     streamType === 'audio_output' || streamType === 'video_output' ? when_defined(statsReport.jitter, 0) * 1000 : null
-                    };
+                    }, statsReport.type, streamType);
+                }
+            } else if(statsReport.type === 'VideoBwe') {
+                if(streamType === 'video_bandwidth') {
+                    extractedStats = new MediaBWEForVideoStats({
+                        timestamp:              timestamp,
+                        encBitrate:             statsReport.googActualEncBitrate,
+                        availReceiveBW:         statsReport.googAvailableReceiveBandwidth,
+                        availSendBW:            statsReport.googAvailableSendBandwidth,
+                        bucketDelay:            statsReport.googBucketDelay,
+                        retransmitBitrate:      statsReport.googRetransmitBitrate,
+                        targetEncBitrate:       statsReport.googTargetEncBitrate,
+                        transmitBitrate:        statsReport.googTransmitBitrate
+                    }, statsReport.type, streamType);
                 }
             }
         }
     }
+    return extractedStats ? extractedStats : null;
+}
 
-    return extractedStats ? new MediaRtpStats(extractedStats, statsReport.type, streamType) : null;
+/**
+* Basic statistics object, represents core properties of any statistic
+*/
+class BaseStats {
+    constructor(paramsIn, statsReportType, streamType) {
```

Comment:
*nitpick* `constructor(params = {}, statsReportType, streamType) {` would remove the need for the `const params = paramsIn || {};` below.